init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/model/Site.java

```diff
@@ -0,0 +1,30 @@
+package uk.ac.ox.ndph.mts.init_service.model;
```

Comment:
does these models should be under the init service? they are later needed in the other micro services?
probably should be in a common location

Reply:
I can return the question - in a micro-service based system, do you want to share models or would that cause too much coupling?

Reply:
Its something that Ive been wondering about - do we create a MTS framework jar containing common classes or classes that should act as Abstract/Parent classes to ensure that the app follows the correct framework? I havent got enough microservices experience to answer the question - not sure if duplication is preferred to inheritance in microservices?

Reply:
The 'microservice ideal' is that the code for each service should be completely independent of any other service. This doesn't stop you from having common base classes - but they should be very generic and independent of any particular service API (because otherwise you have coupled the code in one service to the code in another service, via the base library). We discussed previously having a library of common code which we could reuse as a Maven dependency - I'm not sure if that is up yet. Exactly what is common is a tricky decision. Often you can avoid having to (e.g.) reproduce the model classes of another API by just having better API calls in the first place. I think we don't worry about this for now, at the end of the pilot we will have implemented several different microservices, probably in several different ways, and we can review then what is the appropriate method, and what should be in a common library.

Reply:
I agree with @alexbaxter wrote in general.
Looking through the web, different opinions on sharing model classes in micro-services...